### PR TITLE
Add the cert-manager CSI driver

### LINKED ISSUE #950: [ADDON] Cert Manager CSI driver
csi-driver is a Container Storage Interface (CSI) driver plugin for Kubernetes to work along cert-manager. The goal for this plugin is to seamlessly request and mount certificate key pairs to pods. This is useful for facilitating mTLS, or otherwise securing connections of pods with guaranteed present certificates whilst having all of the features that cert-manager provides. See https://cert-manager.io/docs/projects/csi-driver/

### PR DESCRIPTION
**Pull request by bobdoah**
The cert-manager CSI driver can be used to generate and attach certificates and private keys directly to pods.

### What does this PR do?

Add the Cert Manager CSI driver. Closes #950 

### Motivation

I would like to use this, with Cert Manager as it makes generating and mounting certificates for mTLS very easy.

### More

- [x] Yes, I have tested the PR using my local account setup (Provide any test evidence report under Additional Notes)
- [x] Yes, I have added a new example under [examples](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples) to support my PR
- [x] Yes, I have created another PR for add-ons under [add-ons](https://github.com/aws-samples/eks-blueprints-add-ons) repo (if applicable)
- [x] Yes, I have updated the [docs](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/docs) for this feature
- [x] Yes, I ran `pre-commit run -a` with this PR

**Note**: Not all the PRs require a new example and/or doc page. In general:
- Use an existing example when possible to demonstrate a new addons usage
- A new docs page under `docs/add-ons/*` is required for new a new addon

### For Moderators

- [ ] E2E Test successfully complete before merge?

### CODE REVIEW

**Review by askulkarni2 — COMMENTED** [review]
@bobdoah PR is looking good, minor typos in the doc.

For future, we would love to see an issue first for any significant work - we would hate for your time to be wasted. I've opened #442 for now.

There i

**Comment by askulkarni2** [comment]
@bobdoah any thoughts on updating the [ACM PCA example](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples/tls-with-aws-pca-issuer) we have to show the CSI driver in use?

File: docs/add-ons/cert-manager-csi-driver.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# cert-manager-csi-driver
+
+ver is a Container Storage Interface (CSI) driver plugin for Kubernetes to work along cert-manager. The goal for this plugin is to seamlessly request and mount certificate key pairs to pods. This is useful for facilitating mTLS, or otherwise securing connections of pods with guaranteed present certificates whilst having all of the features that cert-manager provides.
```
[the remark this answers is not in the document]
Comment (bobdoah) [reply docs/add-ons/cert-manager-csi-driver.md]: Yeah, fixed it now.

**Comment by bobdoah** [comment]
> @bobdoah any thoughts on updating the [ACM PCA example](https://github.com/aws-ia/terraform-aws-eks-blueprints/tree/main/examples/tls-with-aws-pca-issuer) we have to show the CSI driver in use?

@askulkarni2 I updated the `eks-cluster-with-new-vpc`, but can update that example too, if desired.

**Review by askulkarni2 — APPROVED** [review]
LGTM! Thanks for the PR.

**Pull request merged**